apps/backend/src/statusActionEngine/emailActionHandler.ts

```diff
@@ -21,203 +22,379 @@ import {
-        }
+      if (recipientWithTemplate.combineEmails) {
+        uoRecipient = [
+          {
+            id: recipientWithTemplate.recipient.name,
+            email: userOfficeEmail,
+            proposals: proposals,
+            template: recipientWithTemplate.emailTemplate.id,
+          },
+        ];
+      } else {
+        const usersDataSource: UserDataSource = container.resolve(
+          Tokens.UserDataSource
+        );
+
+        const recipientPromises = proposals.map(async (proposal) => ({
+          id: recipientWithTemplate.recipient.name,
+          email: userOfficeEmail,
+          proposals: [proposal],
+          template: recipientWithTemplate.emailTemplate.id,
+          instruments: await instrumentDataSource.getInstrumentsByProposalPk(
+            proposal.primaryKey
+          ),
+          pi:
+            (await usersDataSource.getBasicUserInfo(proposal.proposerId)) ||
+            null,
+          coProposers:
+            (await usersDataSource.getProposalUsers(proposal.primaryKey)) ||
+            null,
+        }));
+
+        uoRecipient = await Promise.all(recipientPromises);
+      }
 
-        default:
-          break;
+      await sendMail(
+        uoRecipient,
+        statusActionLogger({
+          connectionId: proposalStatusAction.connectionId,
+          actionId: proposalStatusAction.actionId,
+          statusActionsLogId,
+          emailStatusActionRecipient: EmailStatusActionRecipients.USER_OFFICE,
+          proposalPks,
+        }),
+        statusActionsLogId,
+        loggedInUserId
+      );
+
+      break;
+    }
+
+    case EmailStatusActionRecipients.OTHER: {
+      if (!recipientWithTemplate.otherRecipientEmails?.length) {
+        logger.logError(
+          `Could not execute status action email because no ${EmailStatusActionRecipients.OTHER} recipience set on proposals`,
+          { ...proposalPks }
+        );
+        break;
       }
-    })
-  );
+
+      const otherRecipients: EmailReadyType[] =
+        recipientWithTemplate.otherRecipientEmails.map((email) => ({
+          id: recipientWithTemplate.recipient.name,
+          email: email,
+          proposals: proposals,
+          template: recipientWithTemplate.emailTemplate.id,
+        }));
+
+      await sendMail(
+        otherRecipients,
+        statusActionLogger({
+          connectionId: proposalStatusAction.connectionId,
+          actionId: proposalStatusAction.actionId,
+          emailStatusActionRecipient: EmailStatusActionRecipients.OTHER,
+          proposalPks,
+          statusActionsLogId,
+        }),
+        statusActionsLogId,
+        loggedInUserId
+      );
+
+      break;
+    }
+
+    default:
+      break;
+  }
 };
 
-const sendMail = async (recipientsWithData: EmailReadyType[]) => {
+const sendMail = async (
+  recipientsWithData: EmailReadyType[],
+  statusActionLogger: (
+    actionSuccessful: boolean,
+    message: string
+  ) => Promise<void>,
+  statusActionsLogId?: number | null,
```

Comment:
I would recommend passing option arguments `successfulMessage` and `failMessage` instead of `statusActionsLogId`. This way, `sendEmail` need not know anything about the Status Action Log Feature.